### DISCUSSION TITLE
Error 429

### DISCUSSION THREAD

**Discussion by shiftersbox** [polonel/trudesk#571] [Q&A]
Anybody encountered this error?

My team (4 of us) was trying to login and got this. I thought it was because of the (almost) simultaneous login but I tried it on another remote site via anydesk, got the same error.

How can we fix this? TIA.
<img width="398" alt="Screen Shot 2022-10-03 at 11 15 57 PM" src="https://user-images.githubusercontent.com/114937288/193613690-4b9a10e7-1d22-405d-ae47-897118b2e71c.png">

**Comment by polonel** [comment]
This is the rate limiter kicking in for failed login attempts. Depending on your reverse proxy setup, you can set a couple of env variables to force the rate limiter to limit based on the username and not just IP.

```
USE_XFORWARDIP
USE_USERRATELIMIT
```

Please see [https://github.com/polonel/trudesk/blob/master/src/controllers/main.js#L110-L116](https://github.com/polonel/trudesk/blob/master/src/controllers/main.js#L110-L116) for more information.

**Reply by shiftersbox** [reply]
Thanks @polonel for the quick reply.

I think I could just disable this limits just for internal team use. Could you guide me on how I could edit this on my end? I have trudesk on docker. TIA.

**Reply by polonel** [reply]
Because there is no flag to disable the rate limiter, you'll have to edit the above `main.js` file and then rebuild the docker image. 
The `Dockerfile` is provided, so should be just a simple 
```
cd trudesk
docker build -t polonel/trudesk:custom .
```

Then change your docker command to point to the new image.

**Reply by shiftersbox** [reply]
awesome! thanks!

**Comment by GyVVyG** [comment]
Sorry for the hijack but how do you remove the blacklisted IP? This happened to us this morning after a couple of weeks no problems and can't find any way to bypass.

v1.2.6 install via curled install script. Thanks

**Reply by shiftersbox** [reply]
as mentioned by @polonel you have to edit the main.js file.
